tests/components/wiz_light/test_config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""Test the WiZ Light config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.wiz_light.config_flow import (
+    WizLightConnectionError,
+    WizLightTimeOutError,
+)
+from homeassistant.components.wiz_light.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME
+
+from tests.common import MockConfigEntry
+
+FAKE_BULB_CONFIG = '{"method":"getSystemConfig","env":"pro","result":\
+    {"mac":"ABCABCABCABC",\
+    "homeId":653906,\
+    "roomId":989983,\
+    "moduleName":"ESP_0711_STR",\
+    "fwVersion":"1.21.0",\
+    "groupId":0,"drvConf":[20,2],\
+    "ewf":[255,0,255,255,0,0,0],\
+    "ewfHex":"ff00ffff000000",\
+    "ping":0}}'
+
+TEST_SYSTEM_INFO = {"id": "ABCABCABCABC", "name": "Test Bulb"}
+
+
+TEST_CONNECTION = {CONF_HOST: "1.1.1.1", CONF_NAME: "Test Bulb"}
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+    # Patch booth functions from __init__ with true.
+    with patch(
+        "homeassistant.components.wiz_light.wizlight.getBulbConfig",
+        return_value=FAKE_BULB_CONFIG,
+    ), patch(
+        "homeassistant.components.wiz_light.wizlight.getMac",
+        return_value="ABCABCABCABC",
+    ), patch(
+        "homeassistant.components.wiz_light.async_setup",
+        return_value=True,
+    ) as mock_setup, patch(
+        "homeassistant.components.wiz_light.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            TEST_CONNECTION,
+        )
```

Comment:
I've received feedback in the past to not mock out methods related to the integration and instead mock out the device library. If you were to patch `homeassistant.components.wiz_light.wizlight` would that avoid needing to patch the rest?

Reply:
This was also my first idea but it seems to be difficult to mock the lib  because of the return value of the UDP connection. But in the future this will be the final step to prevent the big "patch stacking".

Reply:
I'm probably missing a key detail about the UDP connection. Could you point out which return value? Would something like this work?

```python
from pywizlight import wizlight
from unittest.mock import create_autospec
...
mock_bulb = create_autospec(wizlight, instance=True)
mock_bulb.getBulbConfig.return_value = FAKE_BULB_CONFIG
mock_bulb.getMac.return_value = 'ABCABCABCABC'
with patch("homeassistant.components.wiz_light.wizlight", return_value=mock_bulb):
    result2 = await hass.config_entries.flow.async_configure(
    ...
```

Reply:
Great, this can work. I have to dig in deeper into that topic because a test for the `light.py` should also be created.
But thank you for the feedback. I'm not a expert in writing tests 😉 ... but the `create_autospec` should be the key.